src/main/java/com/miquido/validoctor/reducerrule/ReducerRule.java

```diff
@@ -50,8 +51,19 @@ public boolean test(PatientType obj) {
   }
 
   @Override
-  public Ailment getAilment() {
-    return rule.getAilment();
+  public Ailment apply(PatientType obj) {
+    PropertyType sum = getters.stream().map(getter -> getter.apply(obj)).reduce(reducer).orElse(null);
```

Comment:
Due to this line, every rule predicate should start with the null check (it's a common approach in javax.validation) and you should add it to documentation.